### PR TITLE
Fix data source option for Prometheus web API in exporting configuration

### LINKED ISSUE #10392: Setting 'data source' in exporting.conf for prometheus doesn't seem to be honored
##### Bug report summary
We're attempting to set 'data source = as collected` in our `[prometheus:exporter]` but the value doesn't seem to be honored.

##### OS / Environment
```
CentOS Linux release 7.8.2003
```

##### Netdata version
```
netdata v1.26.0
```

##### Component Name
Prometheus in exporting.conf

##### Steps To Reproduce
1 Configure `exporting.conf` like:
```
[prometheus:exporter]
    data source = as collected
    send names instead of ids = yes
    send configured labels = yes
    send automatic labels = no
    send charts matching = *
    send hosts matching = localhost *
    prefix = netdata
```
2 Restart netdata.
3 Navigate to prometheus metrics on the host.
```
http://10.20.6.23:19999/api/v1/allmetrics?format=prometheus
```

##### Expected behavior
The prometheus metrics should have a similar output as if I appended a `source` parameter to the URL.  
```
http://10.20.6.23:19999/api/v1/allmetrics?format=prometheus&source=as-collected
```
But the metrics still have an appended `__average` name.

### PR DESCRIPTION
**Pull request by vlvkobal**
##### Summary
Fixes #10392

##### Component Name
Exporting engine

##### Test Plan
Set `data source = raw` in the `[prometheus:exporter]` section of `exporting.conf`. Check the output of `http://localhost:19999/api/v1/allmetrics?format=prometheus` - it should not append `_average` to all metrics.

### CODE REVIEW

**Review by thiagoftsm — APPROVED** [review]
When I set  `data source = raw` inside `exporting.conf` I got this:

```
netdata_host_tags{label1="Catalina"} 1 1608053165613
my_netdata_system_cpu_total{chart="system.cpu",family="cpu",dimension="guest_nice"} 0 1608053164841
my_netdata_system_cpu_total{chart="system.cpu",family="cpu",dimension="guest"} 0 1608053164841
```

and after to comment the option, I got this:

```
netdata_host_tags{label1="Catalina"} 1 1608053095461
my_netdata_system_cpu_percentage_average{chart="system.cpu",family="cpu",dimension="guest_nice"} 0.0000000 1608053092000
my_netdata_system_cpu_percentage_average{chart="system.cpu",family="cpu",dimension="guest"} 0.0000000 1608053092000
```

I ran the first tests with `data  source = average` commented  inside `[backends]`.

Finally I ran another test with `data source = raw` inside `[backends]` and `exporting.conf` without `data source` and I also got the expected result.

LGTM!

File: exporting/read_config.c [hunk 1]
```diff
@@ -238,8 +238,11 @@ struct engine *read_exporting_config()
         prometheus_exporter_instance->config.update_every =
             prometheus_config_get_number(EXPORTING_UPDATE_EVERY_OPTION_NAME, EXPORTING_UPDATE_EVERY_DEFAULT);
 
-        while(!global_backend_source)
-            usleep(100);
+        for (int retries = 0; !global_backend_source && retries < 1000; retries++)
+            usleep(10000);
```
Comment (thiagoftsm) [inline exporting/read_config.c:242]: Please, instead to use `usleep` use `sleep_usec(usec_t usec)`.

Comment (vlvkobal) [reply exporting/read_config.c]: Fixed.

**Review by thiagoftsm — DISMISSED** [review]

File: exporting/read_config.c [hunk 2]
```diff
@@ -238,9 +238,17 @@ struct engine *read_exporting_config()
         prometheus_exporter_instance->config.update_every =
             prometheus_config_get_number(EXPORTING_UPDATE_EVERY_OPTION_NAME, EXPORTING_UPDATE_EVERY_DEFAULT);
 
-        prometheus_exporter_instance->config.options |=
-            global_backend_options &
-            (EXPORTING_SOURCE_DATA_AS_COLLECTED | EXPORTING_SOURCE_DATA_AVERAGE | EXPORTING_SOURCE_DATA_SUM);
+        for (int retries = 0; !global_backend_source && retries < 1000; retries++)
```
Comment (ilyam8) [inline exporting/read_config.c:241]: add a comment or todo?

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]

**Pull request merged**